bindings/matrix-sdk-ffi/src/room_list.rs

```diff
@@ -20,11 +20,18 @@ use crate::{
 #[uniffi::export]
 impl Client {
     /// Get a new `RoomList` instance.
-    pub fn room_list(&self) -> Result<Arc<RoomList>, RoomListError> {
+    ///
+    /// If `with_encryption` is set to true, then the underlying sliding sync
```

Comment:
So this is one of the things I really feel strongly about. Sometime ago we removed the ability of users to disable E2EE support in the sliding sync config because people tend to misconfigure it or just forget to enable it.

Now we are reverting back to giving users the power to mess up their E2EE support.

E2EE should work transparently and should be hard, or ideally impossible, to misconfigure and disable. If people don't want to have E2EE support, we have a feature flag for this.

I mentioned this in the Matrix room, the public API of the `RoomList` or of the `SlidingSync` `Client` should not have changed with this PR.

The only public API change should have been a new convenience `NotificationClient` which does the limited amount of sync requests that the NSE process needs to do to fetch data to render the notification.

Reply:
So, just to explain the situation a bit better. In the case of ElementX iOS, in the WIP branch, we unconditionally spawn two sliding syncs:

1. one RoomList that has `with_encryption` set to false
2. one with `Client::main_encryption_sync`, which creates an `EncryptionSync`.

With this, the two sync loops are already split from each other. You are right that it's an issue if one embedder used the RoomList with `with_encryption = false` and then doesn't spawn the `EncryptionSync`. We can and should make a better API for this, but I'm unclear at which level (sdk-ui? FFI only?), or we could put one `EncryptionSync` within the `RoomList` itself (either in sdk-ui or FFI, again), etc. There are quite a few possible design choices here, and I'm happy to implement now, or in a follow-up PR, what the most appropriate is.

In the NSE extension process for notifications, we only create one `EncryptionSync` via `Client::notification_encryption_sync`, that runs a fixed amount of iterations (right now 2). The idea is to introduce the `NotificationClient` you've talked about, that would automatically run the `EncryptionSync` under the hood, if needs be, and that avoid letting the [embedder do that by themselves](https://github.com/vector-im/element-x-ios/pull/1082/files#diff-1ff0fcebaadd2768e042e7bbafc1321cc5a73089ac1692dc3749492fa2881d91R77-R86).

What are your thoughts, should we do the unified API first, or can we merge this ahead?

Reply:
> What are your thoughts, should we do the unified API first, or can we merge this ahead?

We can merge this and rework the public API in a later PR.